tests/e2e_common_endpoints.py

```diff
@@ -266,23 +270,27 @@ def run_large_message_test(
             )
 
         header_counts = [
+            args.max_http_headers_count - 10,
             args.max_http_headers_count - 1,
             args.max_http_headers_count,
             args.max_http_headers_count + 1,
+            args.max_http_headers_count + 10,
         ]
 
+        # Note: infra generally inserts extra headers (eg, content type and length, user-agent, accept)
+        extra_headers_count = (
+            infra.clients.CCFClient.default_impl_type.extra_headers_count()
+        )
         for s in header_counts:
-            LOG.info("Verifying on cap on max headers count")
-            headers = {f"header-{h}": str(h) for h in range(s)}
-            # Note: infra adds 2 extra headers (content type and length)
-            extra_headers_count = 2
+            LOG.info(f"Verifying on cap on max headers count, sending {s} headers")
+            headers = {f"header-{h}": str(h) for h in range(s - extra_headers_count)}
             run_large_message_test(
                 c,
-                args.max_http_headers_count - extra_headers_count,
+                args.max_http_headers_count,
                 http.HTTPStatus.REQUEST_HEADER_FIELDS_TOO_LARGE,
                 "RequestHeaderTooLarge",
                 "request_header_too_large",
-                len(headers),
+                len(headers) + extra_headers_count,
                 headers=headers,
             )
```

Comment:
The count and logic here were wrong. `requests` may have added 2 headers, but `httpx` adds 5. We were saying something like "the cap is 254, and we're sending 255 headers", which always went down the "this should be an error" path, and _would_ always return an error for a different reason - because we were really sending 259 (>256) headers.

The new call is "the cap is 256, and this request will contain `s` headers (even though `headers` itself may contain fewer)".